distributed/deploy/tests/test_spec_cluster.py

```diff
@@ -209,11 +209,9 @@ async def test_restart(cleanup):
                 cluster.scale(2)
                 await cluster
                 assert len(cluster.workers) == 2
-
                 await client.restart()
-                await asyncio.sleep(3)
-
-                assert len(cluster.workers) == 2
+                while len(cluster.workers) != 2:
+                    await asyncio.sleep(0.5)
 
```

Comment:
Thanks for doing this.  Explicit sleep calls are the bane of CI. 

Question, is there anything here to cause this test to time out if it fails?  If not, this is the sort of thing that can cause CI to hang without feedback.  pytest-asyncio might do this for us.  I'm not sure.  if not then we might want to verify that `time() < start + 10` or something.

Reply:
Ah, I see in setup.cfg that you've added a timeout option.  Great.

Reply:
pytest-timeout will kill off the whole test suite (not just the single test) after 5 minutes

Reply:
@mrocklin I didn't add it; I just moved it from the command-line arguments to setup.cfg

Reply:
added individual timeout to the test